support/maps/maps.go

```diff
@@ -0,0 +1,278 @@
+package maps
+
+import (
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/gookit/goutil/maputil"
+	"github.com/gookit/goutil/reflects"
+)
+
+const (
+	Wildcard = "*"
+	PathSep  = "."
+)
+
+// Add an element to a map using “dot” notation if it doesn't exist.
+func Add(mp *map[string]any, k string, v any) error {
+	if _, ok := maputil.GetByPath(k, *mp); ok {
+		return nil
+	}
+
+	return Set(mp, k, v)
+}
+
+// Dot flattens a map using dot notation.
+func Dot(mp map[string]any) map[string]any {
+	return maputil.Flatten(mp)
+}
+
+// Exists checks if the given key exists in the provided map (only top level).
+func Exists[K comparable, V any](mp map[K]V, key K) bool {
+	_, ok := mp[key]
+	return ok
+}
+
+// Forget removes a given key or keys from the provided map.
+func Forget[K comparable, V any](mp map[K]V, keys ...K) {
+	for _, key := range keys {
+		if _, ok := mp[key]; ok {
+			delete(mp, key)
+			continue
+		}
+
+		if _, ok := any(key).(string); !ok {
+			continue
+		}
+
+		deleteByPathKeys(mp, mp, strings.Split(any(key).(string), PathSep))
+	}
+}
+
+// Get an item from an object using "dot" notation.
+func Get(mp map[string]any, key string, defaults ...any) any {
+	val, ok := maputil.GetByPath(key, mp)
+
+	if !ok && len(defaults) > 0 {
+		return defaults[0]
+	}
+
+	return val
+}
+
+// Has checks if the given key or keys exist in the provided map.
+func Has[K comparable, V any](mp map[K]V, keys ...K) bool {
+	if len(keys) == 0 || len(mp) == 0 {
+		return false
+	}
+
+	for _, key := range keys {
+		if _, ok := any(key).(string); ok {
+			_, ok := maputil.GetByPath(any(key).(string), any(mp).(map[string]any))
+			if !ok {
+				return false
+			}
+
+			continue
+		}
+
+		if !Exists(mp, key) {
+			return false
+		}
+	}
+
+	return true
+}
+
+// HasAny checks if the given key or keys exist in the provided map.
+func HasAny[K comparable, V any](mp map[K]V, keys ...K) bool {
+	for _, key := range keys {
+		if Has(mp, key) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// Only returns the items in the map with the specified keys.
+func Only[K comparable, V any](mp map[K]V, keys ...K) map[K]V {
+	result := make(map[K]V)
+	for _, key := range keys {
+		if Exists(mp, key) {
+			result[key] = mp[key]
+		}
+	}
+
+	return result
+}
+
+// Pull returns a new map with the specified keys removed.
+func Pull[K comparable, V any](mp map[K]V, key K, def ...any) any {
```

Comment:
The action is different with `Arr::pull`:

<img width="723" alt="image" src="https://github.com/goravel/framework/assets/24771476/2c63966e-61d5-42bd-b074-fe074631c038">

Reply:
it's same.